src/signed.rs

```diff
@@ -1040,12 +956,237 @@ macro_rules! int_impl {
+                /// ```
+                #[inline]
+                #[must_use = "this returns the result of the operation, without modifying the original"]
+                pub const fn overflowing_mul(self, rhs: Self) -> (Self, bool) {
+                    let (wrapping_product, overflow) = if (BITS << 1) <= (core::mem::size_of::<$type>() << 3) {
+                        // We have half the bits (e.g. i4 * i4) of the base type, so we can't overflow the base type.
+                        // `wrapping_mul` likely provides the best performance on all CPUs.
+                        (self.value().wrapping_mul(rhs.value()), false)
+                    } else {
+                        // We have more than half the bits (e.g. i6 * i6)
+                        self.value().overflowing_mul(rhs.value())
+                    };
+
+                    let value = (wrapping_product << Self::UNUSED_BITS) >> Self::UNUSED_BITS;
+                    let overflow2 = value != wrapping_product;
+                    (Self { value }, overflow || overflow2)
+                }
+
+                /// Calculates the divisor when `self` is divided by `rhs`.
+                ///
+                /// Returns a tuple of the divisor along with a boolean indicating whether an arithmetic
+                /// overflow would occur. If an overflow would occur then self is returned.
+                ///
+                /// # Panics
+                ///
+                /// This function will panic if `rhs` is zero.
+                ///
+                /// # Examples
+                ///
+                /// Basic usage:
+                ///
+                #[doc = concat!(" ```", $doctest_attr)]
+                /// # use arbitrary_int::prelude::*;
+                /// assert_eq!(i14::new(5).overflowing_div(i14::new(2)), (i14::new(2), false));
+                /// assert_eq!(i14::MIN.overflowing_div(i14::new(-1)), (i14::MIN, true));
+                /// ```
+                #[inline]
+                #[must_use = "this returns the result of the operation, without modifying the original"]
+                pub const fn overflowing_div(self, rhs: Self) -> (Self, bool) {
+                    let (value, overflow) = if Self::UNUSED_BITS == 0 {
+                        // We are something like a Int::<i8; 8>, we can fallback to the base implementation.
+                        // This is very unlikely to happen in practice, but checking allows us to use
+                        // `wrapping_div` instead of `overflowing_div` in the common case, which is faster.
+                        self.value().overflowing_div(rhs.value())
+                    } else {
+                        // We're dealing with fewer bits than the underlying type (e.g. i7).
+                        // That means the division can never overflow the underlying type.
+                        let dividend = self.value().wrapping_div(rhs.value());
```

Comment:
being pedantic here, but this should be quotient (or division_result). `self.value()` is the dividend